lib/web/join_tokens.go

```diff
@@ -265,9 +274,12 @@ func (h *Handler) getAppJoinScriptHandle(w http.ResponseWriter, r *http.Request,
 func (h *Handler) getDatabaseJoinScriptHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params) (interface{}, error) {
 	scripts.SetScriptHeaders(w.Header())
 
+	useCloudStableChannelRepo := h.ClusterFeatures.AutomaticUpgrades && h.ClusterFeatures.Cloud
```

Comment:
Why do we have `&& h.ClusterFeatures.Cloud` check here? Can we omit it, just checking the auto-upgrades flag should be enough? We want to make this work for self-hosted clusters as well.

Reply:
The channel name is `stable/cloud`
Do we want other non-cloud/self-hosted clusters to follow a `cloud` specific version?

It kind of makes sense for self-hosted to follow a "rolling" release channel, but calling it cloud may sound confusing, because they are not using the cloud offering.

Reply:
Agreed on not having self-hosted users using the `stable/cloud` channel (for helm, deb, yum, ...)

As for supporting self-hosted, it's not clear how we are going to do it
For a complete self-hosted management, users would need to set up the s3 bucket for helm charts, but also deb and yum repos, which seems a bit overkill

Reply:
@marcoandredinis @hugoShaka They will have to host their own version lock server but they won't need to host their own repos. They will just use ours. It's ok if we just tell them to use stable/cloud initially, as long as they control the version and are ok with running a version that follows cloud release schedule it should be fine.

Reply:
Let's keep your implementation as-is for now.